### PR TITLE
add a finiteness check to `small_generating_set`

### LINKED ISSUE #4063: `small_generating_set` for matrix groups benefits from computing the order?
I was very excited to find `small_generating_set` for groups. However, for matrix groups it seems to work better (better := produce a smaller set), if one first computes the order.

Without computing the order, one gets basically the generators back:
```
julia> K, a = cyclotomic_field(5, "a");

julia> G = matrix_group([matrix(K, [-1 0 0 0; -a^3-a^2 1 0 0; 0 0 1 0; 0 0 0 1]),
                         matrix(K, [1 -a^3-a^2 0 0; 0 -1 0 0; 0 1 1 0; 0 0 0 1]),
                         matrix(K, [1 0 0 0; 0 1 1 0; 0 0 -1 0; 0 0 1 1]), 
                         matrix(K, [1 0 0 0; 0 1 0 0; 0 0 1 1; 0 0 0 -1])])
Matrix group of degree 4
  over cyclotomic field of order 5

julia> small_generating_set(G)
4-element Vector{MatrixGroupElem{AbsSimpleNumFieldElem, AbstractAlgebra.Generic.MatSpaceElem{AbsSimpleNumFieldElem}}}:
 [-1 0 0 0; -a^3-a^2 1 0 0; 0 0 1 0; 0 0 0 1]
 [1 0 0 0; 0 1 0 0; 0 0 1 1; 0 0 0 -1]
 [1 0 0 0; 0 1 1 0; 0 0 -1 0; 0 0 1 1]
 [1 -a^3-a^2 0 0; 0 -1 0 0; 0 1 1 0; 0 0 0 1]
```
If one first triggers the order computation, the set is smaller:
```
julia> G = matrix_group([matrix(K, [-1 0 0 0; -a^3-a^2 1 0 0; 0 0 1 0; 0 0 0 1]),
                         matrix(K, [1 -a^3-a^2 0 0; 0 -1 0 0; 0 1 1 0; 0 0 0 1]),
                         matrix(K, [1 0 0 0; 0 1 1 0; 0 0 -1 0; 0 0 1 1]), 
                         matrix(K, [1 0 0 0; 0 1 0 0; 0 0 1 1; 0 0 0 -1])])
Matrix group of degree 4
  over cyclotomic field of order 5

julia> order(G)
14400

julia> small_generating_set(G)
2-element Vector{MatrixGroupElem{AbsSimpleNumFieldElem, AbstractAlgebra.Generic.MatSpaceElem{AbsSimpleNumFieldElem}}}:
 [-a^3-a^2+1 -a^3-a^2+1 1 0; 2*a^3+2*a^2-1 2*a^3+2*a^2-1 a^3+a^2-1 -1; 0 -1 0 0; -2*a^3-2*a^2+2 -3*a^3-3*a^2+2 -2*a^3-2*a^2+1 -a^3-a^2+1]
 [0 -1 0 0; a^3+a^2-1 a^3+a^2-1 a^3+a^2-1 -1; -2*a^3-2*a^2+2 -3*a^3-3*a^2+2 -2*a^3-2*a^2+2 -a^3-a^2+1; 2*a^3+2*a^2-1 2*a^3+2*a^2-2 a^3+a^2-2 a^3+a^2-1]
```
This is of course just one example. Would it be reasonable to always trigger an order computation on the OSCAR side for matrix groups?
CC @ThomasBreuer @fingolfin

### PR DESCRIPTION
**Pull request by ThomasBreuer**
... in the case of a matrix group over an infinite base domain

resolves #4063

### CODE REVIEW

**Review by joschmitt — APPROVED** [review]
Thank you!

**Review by lgoettgens — APPROVED** [review]

**Comment by joschmitt** [comment]
Would it make sense to add something like
```
K, a = cyclotomic_field(5, "a")
G = matrix_group([matrix(K, [-1 0 0 0; -a^3-a^2 1 0 0; 0 0 1 0; 0 0 0 1]),
                  matrix(K, [1 -a^3-a^2 0 0; 0 -1 0 0; 0 1 1 0; 0 0 0 1]),
                  matrix(K, [1 0 0 0; 0 1 1 0; 0 0 -1 0; 0 0 1 1]), 
                  matrix(K, [1 0 0 0; 0 1 0 0; 0 0 1 1; 0 0 0 -1])])
@test length(small_generating_set(G)) == 2
```
to the tests or is the output of `small_generating_set` 'too random' for such a test?

**Comment by ThomasBreuer** [comment]
@joschmitt I am investigating which kind of test makes sense here.

**Comment by ThomasBreuer** [comment]
I have now added a test for the case of a finite matrix group over an infinite base domain.
For infinite groups, currently a GAP error happens (independent of the current change), see gap-system/gap/issues/5790.

File: test/Groups/matrixgroups.jl [hunk 1]
```diff
@@ -153,6 +153,21 @@ end
    end
 end
 
+@testset "small_generating_set" begin
+   inputs = [
+     # finite group over an infinite base domain
+     (matrix(QQ, [0 1; 1 0]), matrix(QQ, [0 1; -1 -1])),
+ #   # infinite group
+ #   (matrix(QQ, [0 -1; 1 0]), matrix(QQ, [0 1; -1 -1])),
+#TODO: do not run into a GAP error message
```
Comment (joschmitt) [inline test/Groups/matrixgroups.jl]:
```suggestion
#TODO: do not run into a GAP error message, see
# https://github.com/gap-system/gap/issues/5790
```

**Pull request merged**